chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -157,81 +157,118 @@ pub struct TransactionVerifierDelta {
     token_issuance_cache: ConsumedTokenIssuanceCache,
 }
 
+/// [`TxIndexCache`] that can be enabled or disabled (using a config).
+pub struct GuardedTxIndexCache {
```

Comment:
The file is becoming large again. Can you please move this to some submodule?

Reply:
I have to say also... I don't like the word "guarded"... because "guarding" is reserved for safety operations. I don't have another word in mind though... how about `MaybeTxIndexCache`? Cheesy but expressive xD.

Reply:
Hmm `OptionalTxIndexCache`?

Reply:
That too is OK with me.